script/refunder.js

```diff
@@ -43,7 +59,27 @@ async function main() {
   // Waiting for the transaction to be mined
   const new_receipt = await new_tx.wait();
   // The transaction is now on chain!
-  console.log(`Mined transaction (see in its corresponding block explorer): ${new_receipt.transactionHash}`);
+  console.log(
+    `Mined transaction (see in its corresponding block explorer): ${new_receipt.transactionHash}`
+  );
+}
+
+async function getEthFlowOrderBytes(tx, ethflow_address) {
+  if (tx.to.toLowerCase() === ethflow_address.toLowerCase()) {
+    return tx.data.substring(10);
+  } else {
+    console.log(
+      "Detected a tx that interacted with the ETHFlow contract indirectly"
+    );
+
+    const create_order_index = tx.data.indexOf(CREATE_ORDER_SELECTOR);
+    if (create_order_index === -1) {
+      throw new Error("createOrder function selector not found in tx.data.");
+    }
+
+    const order_start = create_order_index + CREATE_ORDER_SELECTOR.length;
+    return tx.data.substring(order_start, order_start + ETHFLOW_ORDER_LENGTH);
```

Comment:
This part is unnecessarily fragile. It's possible that the calldata of the tx contains the byte sequence of the function selector by chance (or malicious intent). Then this function would copy nonsense.
IMO the correct way to do things would be to decode the order data from the event and create a new `invalidateOrder` call with that.